libs/vkd3d/command.c

```diff
@@ -11568,18 +11568,21 @@ static void d3d12_command_list_clear_uav(struct d3d12_command_list *list,
             buffer_info.range = args->u.buffer.range;
         }
 
-        miplevel_idx = 0;
         layer_count = 1;
         pipeline = vkd3d_meta_get_clear_buffer_uav_pipeline(&list->device->meta_ops,
                 !args->has_view || args->u.view->format->type == VKD3D_FORMAT_TYPE_UINT,
                 !args->has_view);
         workgroup_size = vkd3d_meta_get_clear_buffer_uav_workgroup_size();
+
+        view_extent.width = resource->desc.Width;
```

Comment:
This seems redundant. In line 11594, full_rect.right is overwritten based on byte_count. Seems slightly misleading to set view_extent.width to anything other than 0 here with a comment saying  it'll be overriden. Another idea might be to pull that branch into here. sampler_feedback_clear is only for images anyway.

Reply:
Cleaned this up a bit.